node_package/src/clientStartup.ts

```diff
@@ -304,12 +304,7 @@ export function clientStartup(context: Context): void {
 
   debugTurbolinks('Adding DOMContentLoaded event to install event listeners.');
 
-  // So  long as the document is not loading, we can assume:
-  // The document has finished loading and the document has been parsed
-  // but sub-resources such as images, stylesheets and frames are still loading.
-  // If lazy asynch loading is used, such as with loadable-components, then the init
-  // function will install some handler that will properly know when to do hyrdation.
-  if (document.readyState !== 'loading') {
+  if (document.readyState === 'complete') {
     window.setTimeout(renderInit);
   } else {
     document.addEventListener('DOMContentLoaded', renderInit);
```

Comment:
From MDN:
1. [`The DOMContentLoaded event fires when the initial HTML document has been completely loaded and parsed, without waiting for stylesheets, images, and subframes to finish loading.`](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event)
2. [`readyState === 'complete': The document and all sub-resources have finished loading. The state indicates that the load event is about to fire.`](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState#readystatechange_as_an_alternative_to_load_event)

The event listener and the if statement correspond to different stages of page load.

Should it be something like the code below?
```
function onPageReady(callback) {
  if (document.readyState === "complete") {
    callback();
  } else {
    document.addEventListener("readystatechange", function onReadyStateChange() {
        onPageReady(callback);
        document.removeEventListener("readystatechange", onReadyStateChange);
    });
  }
}
```

Reply:
Yes, you're right, @Romex91.

Do you think this change could cause any sort of performance regression for our clients?